one_fm/hiring/utils.py

```diff
@@ -355,13 +359,35 @@ def update_erf_close_with(doc):
 @frappe.whitelist()
 def create_salary_structure_assignment(doc, method):
     if doc.job_offer_salary_structure:
+        job_offer = frappe.get_doc("Job Offer", doc.job_offer)
         assignment = frappe.new_doc("Salary Structure Assignment")
         assignment.employee = doc.name
-        assignment.salary_structure = doc.job_offer_salary_structure
```

Comment:
I think this needs to be there and then by condition we can assign the new salary structure